src/pudl/extract/csv.py

```diff
@@ -1,72 +1,121 @@
 """Extractor for CSV data."""
-from csv import DictReader
-from importlib import resources
-from zipfile import ZipFile
-
 import pandas as pd
 
 import pudl.logging_helpers
+from pudl.extract.extractor import GenericExtractor
 
 logger = pudl.logging_helpers.get_logger(__name__)
 
 
-def open_csv_resource(dataset: str, base_filename: str) -> DictReader:
-    """Open the given resource file as :class:`csv.DictReader`.
+class CsvExtractor(GenericExtractor):
+    """Class for extracting dataframes from CSV files.
 
-    Args:
-        dataset: used to load metadata from package_data/{dataset} subdirectory.
-        base_filename: the name of the file in the subdirectory to open.
+    The extraction logic is invoked by calling extract() method of this class.
     """
-    csv_path = resources.files(f"pudl.package_data.{dataset}") / base_filename
-    return DictReader(csv_path.open())
-
 
-def get_table_file_map(dataset: str) -> dict[str, str]:
-    """Return a dictionary of table names and filenames for the dataset.
-
-    Args:
-        dataset: used to load metadata from package_data/{dataset} subdirectory.
-    """
-    return {
-        row["table"]: row["filename"]
-        for row in open_csv_resource(dataset, "table_file_map.csv")
-    }
+    def __init__(self, ds, dataset_name):
+        """Create a new instance of CsvExtractor.
 
+        This can be used for retrieving data from CSV files.
 
-class CsvExtractor:
-    """Generalized class for extracting dataframes from CSV files.
+        Args:
+            ds (datastore.Datastore): An initialized datastore, or subclass
+            dataset_name: Name of the package/dataset to load
+        """
+        self.ds = ds
+        self._dataset_name = dataset_name
 
-    The extraction logic is invoked by calling extract() method of this class.
-    """
+    def extract(self, **partition):
+        """Extracts dataframes.
 
-    def __init__(self, zipfile: ZipFile, table_file_map: dict[str, str]):
-        """Create a new instance of CsvExtractor.
+        Returns dict where keys are page names and values are
+        DataFrames containing data across given years.
 
-        This can be used for retrieving data from CSV files.
+        Args:
+            partitions (list, tuple or string): list of partitions to
+                extract. (Ex: [2009, 2010])
+        """
+        raw_dfs = {}
+        logger.info(f"Extracting {self._dataset_name} data.")
+
+        page = "data"
+        dfs = [
+            pd.DataFrame(),
+        ]
+        # TODO: Use the below commented-out code for consistency? I'm only expecting individual years at this point
+        # for partition in pudl.helpers.iterate_multivalue_dict(**partition):
```

Comment:
This method is still messy and I will definitely clean it up. Here and below there are multiple blocks of code from the Excel extractor that could be worth conforming to, depending on how defensive/sophisicated/conformist we want to be. It could largely get reused/"contracted" in a DRY manner if we decompose/refactor things. Any thoughts on how much to put into making methods/"contracts" conform across extractors of different types?

Reply:
I think there's a lot of potential to condense and refactor things here, if you're up for it! I left some thoughts on how I would go about restructuring this below.